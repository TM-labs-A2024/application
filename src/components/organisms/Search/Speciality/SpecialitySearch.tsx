import {
  Link,
  Button,
  Text,
  Heading,
  FormControl,
  FormLabel,
  Input,
  Stack,
  FormErrorMessage,
  HStack,
  Tag,
  TagLabel,
  TagCloseButton
} from '@chakra-ui/react'
import SearchInputComponent from '@components/atoms/SearchInput'
import EvolutionList from '@components/molecules/EvolutionsList'
import { evolutionTypes } from '@constants/index'
import { FILTERS_APPLIED } from '@constants/index'
import { Evolution, ReactSelectOption } from '@src/types'
import { isIOS, isMobile } from '@utils/index'
import { format } from 'date-fns'
import NextLink from 'next/link'
import React, { useEffect, useState, useCallback } from 'react'
import { Controller, useForm, FieldErrors } from 'react-hook-form'
import { Store } from 'react-notifications-component'
import Select from 'react-select'

import Swimmer from '../../../../../public/static/icons/Swimmer.svg'

const evolutionTypesOptions = evolutionTypes.map(({ id, name }: { id: number; name: string }) => ({
  value: id,
  label: name
}))

type FormData = {
  type: ReactSelectOption | null
  fromDate: string
  toDate: string
}

export default function SpecialitySearch({
  context
}: {
  context: {
    speciality: string | string[] | undefined
    onChange: (value: string) => void
    data: {
      evolutions: Evolution
      orders: Evolution
      tests: Evolution
    }
    matches: string
    fromDate: string
    toDate: string
    type: ReactSelectOption
    setFromDate: React.Dispatch<React.SetStateAction<string>>
    setToDate: React.Dispatch<React.SetStateAction<string>>
    setType: React.Dispatch<React.SetStateAction<ReactSelectOption>>
  }
}) {
  // --- Hooks -----------------------------------------------------------------
  const {
    handleSubmit,
    formState: { errors },
    control,
    register,
    watch,
    setValue
  } = useForm<FormData>()

  const values = watch()
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  const [showFilters, setShowFilters] = useState(false)

  const { fromDate, toDate, type, setFromDate, setToDate, setType, onChange } = context
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  const inputRef = React.createRef<HTMLInputElement>()
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  useEffect(() => {
    if (inputRef.current) {
      inputRef.current.focus()
    }
  }, [inputRef])
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const removeFilters = useCallback(() => {
    setFromDate('')
    setToDate('')
    setType({
      value: 0,
      label: ''
    })
    setValue('fromDate', '')
    setValue('toDate', '')
    setValue('type', null)
    setShowFilters(false)
  }, [setFromDate, setToDate, setType, setValue])

  const onSubmit = (data: FormData) => {
    setFromDate(data.fromDate)
    setToDate(data.toDate)
    setType(data?.type)
    setShowFilters(false)
    Store.addNotification(FILTERS_APPLIED(isMobile(window)))
  }

  const verifyErrors = (errors: FieldErrors<FormData>) => Object.keys(errors).length > 0
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`relative mx-auto block h-screen w-screen overflow-hidden px-8 lg:px-96 ${isIOS() ? 'pt-20' : 'pt-8'}`}
    >
      {showFilters && (
        <>
          <div className="mb-8 flex flex-row items-center justify-between gap-4">
            <Text>Ingresar filtros</Text>
            <Button variant="link" colorScheme="black" fontWeight="normal" onClick={removeFilters}>
              Cancelar
            </Button>
          </div>
          <Heading as="h2" size="md">
            Fechas de evolución
          </Heading>
          <form onSubmit={handleSubmit(onSubmit)} className="h-full w-full">
            <FormControl isInvalid={verifyErrors(errors)} className="relative h-4/5 w-full">
              <div className="my-6 flex w-full flex-row gap-4">
                <div className="flex w-1/2 flex-col">
                  <FormLabel>Desde</FormLabel>
                  <Input
                    id="fromDate"
                    type="date"
                    placeholder="Desde"
                    defaultValue={fromDate}
                    {...register('fromDate', {
                      validate: {
                        dateBefore: (value) =>
                          values.toDate === null ||
                          values.toDate === '' ||
                          value <= values.toDate ||
                          'La fecha de comienzo debe ser menor a la fecha final'
                      }
                    })}
                  />
                  <FormErrorMessage>
                    {errors?.fromDate && errors?.fromDate?.message}
                  </FormErrorMessage>
                </div>
                <div className="flex w-1/2 flex-col">
                  <FormLabel>Hasta</FormLabel>
                  <Input
                    id="toDate"
                    type="date"
                    placeholder="Hasta"
                    defaultValue={toDate}
                    {...register('toDate', {
                      validate: {
                        dateBefore: (value) =>
                          values.fromDate === null ||
                          values.fromDate !== '' ||
                          value >= values.fromDate ||
                          'La fecha final debe ser mayor a la fecha de comienzo'
                      }
                    })}
                  />
                  <FormErrorMessage>{errors?.toDate && errors?.toDate?.message}</FormErrorMessage>
                </div>
              </div>
              <Controller
                control={control}
                name="type"
                render={({ field }) => (
                  <Select id="type" {...field} placeholder="Tipo" options={evolutionTypesOptions} />
                )}
              />
              <Stack spacing={4} mt={4} className="absolute bottom-0 w-full">
                <Button type="submit">Aplicar filtros</Button>
                <Button
                  variant="link"
                  colorScheme="black"
                  fontWeight="normal"
                  onClick={removeFilters}
                >
                  Quitar filtros
                </Button>
              </Stack>
            </FormControl>
          </form>
        </>
      )}
      {!showFilters && (
        <>
          <div className="mb-8 flex flex-row items-center gap-4">
            <SearchInputComponent
              placeholder="ingresar criterio"
              className="w-full"
              onChange={onChange}
              inputRef={inputRef}
            />
            <Link as={NextLink} href={`/especialidad/${context.speciality}`}>
              Cancelar
            </Link>
          </div>
          <div className="w-full overflow-x-scroll">
            <HStack spacing={4} mb={4} className="w-max">
              {fromDate && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>
                    Desde: {format(new Date(fromDate.replace(/-/g, '/')), 'dd/MM/yyyy')}
                  </TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setFromDate('')
                      setValue('fromDate', '')
                    }}
                  />
                </Tag>
              )}
              {toDate && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>
                    Hasta: {format(new Date(toDate.replace(/-/g, '/')), 'dd/MM/yyyy')}
                  </TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setToDate('')
                      setValue('toDate', '')
                    }}
                  />
                </Tag>
              )}
              {type?.label && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>Tipo: {type.label}</TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setType(null)
                      setValue('type', null)
                    }}
                  />
                </Tag>
              )}
            </HStack>
          </div>
          {context.matches !== '' && (
            <Heading as="h2" size="sm" noOfLines={1} className="mb-4">
              {context.matches}
            </Heading>
          )}
          <div className="h-3/4 overflow-scroll">
            <div className="block">
              {context.data.evolutions.length > 0 && <Text>Evoluciones</Text>}
              <EvolutionList evolutions={context.data.evolutions} />
            </div>
            <div className="block">
              {context.data.orders.length > 0 && <Text>Ordenes</Text>}
              <EvolutionList evolutions={context.data.orders} />
            </div>
            <div className="block">
              {context.data.tests.length > 0 && <Text>Análisis</Text>}
              <EvolutionList evolutions={context.data.tests} />
            </div>
          </div>
          <div className="absolute bottom-16 w-full pr-16">
            <Button className="w-full gap-2" onClick={() => setShowFilters(!showFilters)}>
              <Swimmer /> Filtros
            </Button>
          </div>
        </>
      )}
    </div>
  )
}
