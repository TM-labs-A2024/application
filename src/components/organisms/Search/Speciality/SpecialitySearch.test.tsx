import { specialityData } from '@src/constants'
import { render, screen, fireEvent, waitFor } from '@test/utils/index'

import '@testing-library/jest-dom'
import 'react-notifications-component'
import SpecialitySearch from './SpecialitySearch'

jest.mock('react-notifications-component')

const context = {
  goBackRef: '',
  specialityData,
  isNurse: false,
  isPatient: true
}

describe('Organisms > Search > SpecialitySearch test', () => {
  test('The component shows the evolutions which match the entered criteria', () => {
    render(<SpecialitySearch context={context} />)

    const input = screen.getByPlaceholderText('ingresar criterio')
    fireEvent.change(input, { target: { value: 'Evolución 1' } })

    const titleMatcher = (content: string) => content.includes('Evolución 1')
    const titleElement = screen.getByText(titleMatcher)

    expect(titleElement).toBeInTheDocument()
  })

  test('The component shows the filters if the filters button is clicked', () => {
    render(<SpecialitySearch context={context} />)

    const button = screen.getByText('Filtros')
    fireEvent.click(button)

    expect(screen.getByText('Ingresar filtros')).toBeInTheDocument()
    expect(screen.getByText('Desde')).toBeInTheDocument()
    expect(screen.getByText('Hasta')).toBeInTheDocument()
    expect(screen.getByText('Tipo')).toBeInTheDocument()
    expect(screen.getByText('Aplicar filtros')).toBeInTheDocument()
    expect(screen.getByText('Quitar filtros')).toBeInTheDocument()
  })

  test('The component shows the evolutions which match the entered filters and shows the filters on top of the results', async () => {
    render(<SpecialitySearch context={context} />)

    const button = screen.getByText('Filtros')
    fireEvent.click(button)

    const inputFrom = screen.getByPlaceholderText('Desde')
    fireEvent.change(inputFrom, { target: { value: '2024-06-08' } })

    const submitButton = screen.getByText('Aplicar filtros')
    fireEvent.click(submitButton)

    await waitFor(() => {
      expect(screen.getAllByTestId('evolutions-list')).toHaveLength(3)
    })

    expect(screen.getByText('Desde: 08/06/2024')).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<SpecialitySearch context={context} />)

    expect(screen.getByTestId('speciality-search')).toMatchSnapshot()
  })
})
