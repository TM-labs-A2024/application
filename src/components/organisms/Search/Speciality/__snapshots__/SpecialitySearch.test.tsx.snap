// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Search > SpecialitySearch test Matches the snapshot 1`] = `
<div
  class="relative mx-auto block h-screen w-screen overflow-hidden px-8 lg:w-1/2 pt-8"
  data-testid="speciality-search"
>
  <div
    class="mb-8 flex flex-row items-center gap-4"
  >
    <div
      class="w-full"
      data-testid="search-input-component"
    >
      <div
        class="chakra-input__group css-1y0e7gb"
        data-group="true"
      >
        <div
          class="chakra-input__left-element css-gv1iwz"
        >
          <svg
            class="chakra-icon css-13izzit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <input
          class="chakra-input min-h-10 css-qluo93"
          placeholder="ingresar criterio"
        />
      </div>
    </div>
    <a
      class="chakra-link css-spn4bz"
      href=""
    >
      Cancelar
    </a>
  </div>
  <div
    class="w-full overflow-x-scroll"
  >
    <div
      class="chakra-stack w-max css-hrntqi"
    />
  </div>
  <h2
    class="chakra-heading mb-4 css-o8iskg"
  >
    27 resultados
  </h2>
  <div
    class="h-3/4 overflow-scroll"
  >
    <div
      class="block"
    >
      <p
        class="chakra-text css-0"
      >
        Evoluciones
      </p>
      <ul
        class="h-full css-1xa84ef"
        data-testid="evolutions-list"
        role="list"
      >
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/1"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 1: 17, septiembre 2024
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/2"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 2: 17, septiembre 2024
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/3"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 3: 17, septiembre 2024
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/4"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 4: 17, septiembre 1997
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/5"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 5: 17, septiembre 1997
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/6"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 6: 17, septiembre 1997
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/7"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 7: 17, septiembre 1997
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/8"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 8: 17, septiembre 1997
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/evolucion/undefined/9"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Evolución 9: 17, septiembre 1997
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Editado por: Pedro Pérez
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Patología: Dolor
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
      </ul>
    </div>
    <div
      class="block"
    >
      <p
        class="chakra-text css-0"
      >
        Ordenes
      </p>
      <ul
        class="h-full css-1xa84ef"
        data-testid="evolutions-list"
        role="list"
      >
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/1"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 1
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/2"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 2
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/3"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 3
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/4"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 4
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/5"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 5
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/6"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 6
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/7"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 7
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/8"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 8
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/orden/undefined/9"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Orden 9
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
      </ul>
    </div>
    <div
      class="block"
    >
      <p
        class="chakra-text css-0"
      >
        Análisis
      </p>
      <ul
        class="h-full css-1xa84ef"
        data-testid="evolutions-list"
        role="list"
      >
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/1"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 1
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/2"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 2
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/3"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 3
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/4"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 4
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/5"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 5
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/6"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 6
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/7"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 7
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/8"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 8
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
        <div
          class="border-b border-black last:border-transparent"
        >
          <a
            class="chakra-link w-full css-ofs0kf"
            data-testid="evolution-card"
            href="/analisis/undefined/9"
          >
            <div
              class="mb-1 flex flex-row items-center justify-between"
            >
              <div
                class="flex flex-col items-start gap-1"
              >
                <b
                  class="chakra-text css-qqfgvy"
                >
                  Análisis 9
                </b>
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    17, septiembre 1997
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    Agregado por: Pedro Pérez
                  </p>
                </div>
              </div>
              <div />
            </div>
          </a>
        </div>
      </ul>
    </div>
  </div>
  <div
    class="absolute bottom-16 w-full pr-16"
  >
    <button
      class="chakra-button w-full gap-2 css-ez23ye"
      type="button"
    >
      <div />
       Filtros
    </button>
  </div>
</div>
`
